src/routes/safe/components/Settings/Advanced/index.tsx

```diff
@@ -0,0 +1,291 @@
+  const autoColumns = columns.filter(({ custom }) => !custom)
+
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const nonce = useSelector(safeNonceSelector)
+  const granted = useSelector(grantedSelector)
+  const modules = useSelector(safeModulesSelector)
+  const moduleData = modules !== null ? getModuleData(List(modules)) : modules
+
+  const [viewRemoveModuleModal, setViewRemoveModuleModal] = React.useState(false)
+  const hideRemoveModuleModal = () => setViewRemoveModuleModal(false)
+
+  const [selectedModule, setSelectedModule] = React.useState(null)
+  const triggerRemoveSelectedModule = (module: [string, string]): void => {
+    setSelectedModule(module)
+    setViewRemoveModuleModal(true)
+  }
+
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar()
+  const dispatch = useDispatch()
+
+  const removeSelectedModule = async (): Promise<void> => {
+    try {
+      const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+      const [module, prevModule] = selectedModule
+      const txData = safeInstance.contract.methods.disableModule(prevModule, module).encodeABI()
+
+      dispatch(
+        createTransaction({
+          safeAddress,
+          to: safeAddress,
+          valueInWei: '0',
+          txData,
+          notifiedTransaction: TX_NOTIFICATION_TYPES.SETTINGS_CHANGE_TX,
+          enqueueSnackbar,
+          closeSnackbar,
+        }),
+      )
+    } catch (e) {
+      console.error(`failed to remove the module ${selectedModule}`, e.message)
+    }
+  }
+
+  return (
+    <>
+      {/* Nonce */}
+      <Block className={classes.container}>
+        <Title size="xs" withoutMargin>
+          Safe Nonce
+        </Title>
+        <InfoText size="lg">
+          For security reasons, transactions made with the Safe need to be executed in order. The nonce shows you which
+          transaction was executed most recently. You can find the nonce for a transaction in the transaction details.
+        </InfoText>
+        <InfoText color="secondaryLight" size="xl">
+          Current Nonce: <Bold>{nonce}</Bold>
+        </InfoText>
+      </Block>
+
+      {/* Modules */}
+      <Block className={classes.container}>
+        <Title size="xs" withoutMargin>
+          Safe Modules
+        </Title>
+        <InfoText size="lg">
+          Modules allow you to customize the access-control logic of your Safe. Modules are potentially risky, so make
+          sure to only use modules from trusted sources. Learn more about modules{' '}
+          <a
+            href="https://docs.gnosis.io/safe/docs/contracts_architecture/#3-module-management"
+            rel="noopener noreferrer"
+            target="_blank"
+          >
+            here
+          </a>
+          .
+        </InfoText>
+        {moduleData === null ? (
+          <InfoText color="secondaryLight" size="xl">
+            No modules enabled
+          </InfoText>
+        ) : moduleData.size === 0 ? (
```

Comment:
is it possible to refactor this without using nested ternaries?

in general I think this component could be split even further (For example, modal and table could be a separate components)

Reply:
this was still not addressed

Reply:
I refactored this one, but let the ternary, It's more readable now. Despite I'm not a fan of the nested ternaries, I find it quite simple to understand at first sight in this case now.